### PR TITLE
Master GitHub actions

### PR DESCRIPTION
**Pull request by Anupamatd**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            database: 'pgsql'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_311_STABLE'
+            database: 'mariadb'
+            node: '14.15'
+          - php: '7.3'
+            moodle-branch: 'MOODLE_311_STABLE'
+            database: 'pgsql'
+            node: '14.15'
```
Comment (jwalits) [inline .github/workflows/ci.yml:105]: this matrix is quite hard to follow and will grow out of control when more versions are added...maybe consider simplifying that to something like this: https://github.com/catalyst/moodle-auth_saml2/blob/MOODLE_39_STABLE/.github/workflows/39-master.yml#L43-L55

Comment (Anupamatd) [reply .github/workflows/ci.yml]: Hey @jwalits - updated with the given reference. Please review.. Thank you...
